akka-actor/src/main/resources/reference.conf

```diff
@@ -906,5 +906,59 @@ akka {
     }
   }
 
+  coordinated-shutdown {
+    # The timeout that will be used for a phase if not specified with 
+    # 'timeout' in the phase
+    default-timeout = 5 s
+    
+    # Terminate the ActorSystem in the last phase actor-system-terminate.
+    terminate-actor-system = on
+    
+    # Exit the JVM (System.exit(0)) in the last phase actor-system-terminate
+    # if this is set to 'on'. It is done after termination of the 
+    # ActorSystem if terminate-actor-system=on, otherwise it is done 
+    # immediately when the last phase is reached.  
+    exit-jvm = off
+    
+    # Run the coordinated shutdown when the JVM process exits, e.g.
+    # via kill signal (ctrl-c).
+    run-by-jvm-shutdown-hook = on
+    
+    # Each phase is defined as a named config section with the
+    # following optional properties:
+    # - timeout=15s: Override the default-timeout for this phase.
+    # - recover=off: If the phase fails the shutdown is aborted
+    #                and depending phases will not be executed.
+    # dependsOn=[]:  Run the phase after the given phases
+    phases {
+      cluster-sharding-shutdown-region {
+        timeout = 10 s
+      }
+      
+      cluster-leave {
+        depends-on = [cluster-sharding-shutdown-region]
+      }
+      
+      # shutdown cluster singletons
+      cluster-exiting {
+        timeout = 10 s
+        depends-on = [cluster-leave]
+      }
+      
+      cluster-exiting-done {
+        depends-on = [cluster-exiting]
+      }
+      
+      cluster-shutdown {
+        depends-on = [cluster-exiting-done]
+      }
+      
+      # Last phase. See terminate-actor-system and exit-jvm above.
+      actor-system-terminate {
+        timeout = 10 s
+        depends-on = [cluster-shutdown]
+      }
```

Comment:
It is more tricky for HTTP though as there is no single server, there can be many of them (unlike cluster)